var/spack/repos/builtin/packages/gcc/package.py

```diff
@@ -1150,6 +1150,9 @@ def runtime_constraints(cls, *, spec, pkg):
         # The version of gcc-runtime is the same as the %gcc used to "compile" it
         pkg("gcc-runtime").requires(f"@={str(spec.version)}", when=f"%{str(spec)}")
 
+        # If a node used %gcc@X.Y its dependencies must use gcc-runtime@:X.Y
```

Comment:
@:x can include versions larger than @=x

Reply:
I think it's the comment to be imprecise. This is what gets generated:
```
propagate(node(ID, Package), node_compiler_version_satisfies("gcc", ":13.2.0")) :-
  attr("node", node(ID, Package)),
  attr("node_compiler", node(ID, Package), "gcc"),
  attr("node_compiler_version", node(ID, Package), "gcc", "13.2.0"),
  not external(node(ID, Package)),
  not runtime(Package).
```
unless we are arguing about potential 13.2.0.1 versions being added.

Reply:
Any version can be defined in config, no minor/patch required.

Reply:
Can you leave a comment at least so we remember this is wrong, but just so happens to work? It's not a general trick we can copy over to other runtimes if they have a different versioning scheme.

Reply:
are patches potentially incompatible?

Reply:
there is only major.minor really, custom patch version number is set by rhel

Reply:
Note added. I think this logic will change when moving to dependencies. Also, this rule:
```
% If we build packages, the runtime nodes must use an available compiler
1 { node_compiler(PackageNode, CompilerID) : build(PackageNode), not external(PackageNode) } :-
  has_built_packages(),
  runtime(RuntimePackage),
  node_compiler(node(_, RuntimePackage), CompilerID).
```
should mitigate the issue. It won't cover for `foo%gcc@13 ^bar%gcc@13.1` but I guess the cross-section for those specs is not huge?

Reply:
For example rockylinux:9 has GCC `11.4.1` which is not an official release.

We could stick to `@:major.minor` (and pad with 0 if no minor version) to improve reuse...

Reply:
> I meant to say: we could compare on the level of major.minor only.

To do that it's probably better to have a way to express `<=` in versions. The rule in https://github.com/spack/spack/pull/44834#discussion_r1656660605 should cover for most corner cases.